src/repositories/jobRepository.ts

```diff
@@ -21,8 +21,14 @@ export class JobRepository extends BaseRepository {
     this._queueConnector = new SQSConnector(logger, config);
   }
 
-  async updateWithCompletionStatus(id: string, result: any): Promise<boolean> {
-    const query = { _id: id };
+  async updateWithCompletionStatus(
+    id: string | mongodb.ObjectId,
+    result: any,
+    shouldNotifySqs = true
+  ): Promise<boolean> {
+    // Safely convert to object ID
+    const objectId = new mongodb.ObjectId(id);
+    const query = { _id: objectId };
```

Comment:
For some reason, this expected a string, but queries should typically expect an ObjectId. This change should help enforce valid IDs to be used while ensuring jobs can be queried successfully regardless of whether the ID is in a string or ObjectId format